python/Scripts/generateShader.py

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python
+'''
+Utility to generate the shader for materials found in a MaterialX document. One file will be generated
```

Comment:
I really like the idea of adding this new script to the codebase.  One minor adjustment I would recommend is to name it "generateshader.py", for letter-case consistency with our existing scripts.